middle_end/flambda/remove_unused_arguments.ml

```diff
@@ -188,15 +188,16 @@ let should_split_only_specialised_args
       Variable.Set.is_empty
         (Find_recursive_functions.in_function_declarations fun_decls ~backend)
     in
-    let number_of_non_stub_functions =
-      Variable.Map.cardinal
-        (Variable.Map.filter (fun _ { Flambda.stub } -> not stub)
-           fun_decls.funs)
+    let non_stub_functions =
+      Variable.Map.filter (fun _ { Flambda.stub } -> not stub)
+        fun_decls.funs
     in
     (* CR-soon lwhite: this criteria could use some justification.
        mshinwell: pchambart cannot remember how these criteria arose,
        but we're going to leave this as-is for 4.03. *)
-    no_recursive_functions && (number_of_non_stub_functions <= 1)
+    no_recursive_functions &&
+    (Variable.Map.is_empty non_stub_functions ||
+     Variable.Map.is_singleton non_stub_functions)
```

Comment:
During review, @lthls argued that it would be better not to change the existing code here. @kit-ty-kate: can you revert this thunk of the patch?

Reply:
done